ldi-core/ldi-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/rdf/formatter/JsonLdFrameWriter.java

```diff
@@ -1,24 +1,27 @@
 package be.vlaanderen.informatievlaanderen.ldes.ldi.rdf.formatter;
 
-import org.apache.jena.atlas.json.JSON;
-import org.apache.jena.atlas.json.JsonObject;
+import com.apicatalog.jsonld.JsonLd;
+import com.apicatalog.jsonld.JsonLdError;
+import com.apicatalog.jsonld.document.JsonDocument;
+import jakarta.json.JsonObject;
 import org.apache.jena.rdf.model.Model;
-import org.apache.jena.riot.*;
-import org.apache.jena.riot.writer.JsonLD10Writer;
-import org.apache.jena.sparql.util.Context;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFFormat;
+import org.apache.jena.riot.RDFWriter;
+import org.apache.jena.riot.RDFWriterBuilder;
 
 import java.io.OutputStream;
-
-import com.github.jsonldjava.core.JsonLdOptions;
+import java.io.StringReader;
 
 import static be.vlaanderen.informatievlaanderen.ldes.ldi.rdf.formatter.PrefixAdder.addPrefixesToModel;
 
 public class JsonLdFrameWriter implements LdiRdfWriter {
 	private final RDFWriterBuilder rdfWriter;
+	private final JsonDocument frame;
 
-	public JsonLdFrameWriter(LdiRdfWriterProperties properties) {
-		String frame = properties.getJsonLdFrame();
-		this.rdfWriter = RDFWriter.create().context(getFramedContext(frame)).format(RDFFormat.JSONLD10_FRAME_PRETTY);
+	public JsonLdFrameWriter(LdiRdfWriterProperties properties) throws JsonLdError {
```

Comment:
I don't like this. A constructor should not throw exceptions imo, but we have had this logic in constructor discussion before :)